### PR TITLE
[doc] Rename MLBuffer => MLTensor for WebNN EP

### PR DESCRIPTION
**Pull request by Honry**

### CODE REVIEW

**Comment by Honry** [comment]
@fdwr, @guschmue, @egalli, PTAL, thanks!

File: docs/tutorials/web/ep-webnn.md [hunk 1]
```diff
@@ -74,59 +74,59 @@ To use WebNN EP, you just need to make 3 small changes:
...
+If your input data is a WebNN storage MLTensor, you can create a MLTensor tensor and use it as input tensor:
 
 ```js
 const mlContext = await navigator.ml.createContext({deviceType, ...});
-const inputMLBuffer = await mlContext.createBuffer({
+const inputMLTensor = await mlContext.createTensor({
   dataType: 'float32',
   dimensions: [1, 3, 224, 224],
-  usage: MLBufferUsage.WRITE_TO,
+  usage: MLTensorUsage.WRITE_TO,
```
Comment (fdwr) [inline docs/tutorials/web/ep-webnn.md:102]: FYI, I think this was renamed to `MLTensorUsage.WRITE` (and `READ_FROM` to `READ`) per https://github.com/webmachinelearning/webnn/pull/754/files, but if we just want to keep it to the rename, that's fine.

File: docs/tutorials/web/ep-webnn.md [hunk 2]
```diff
@@ -140,25 +140,25 @@ const results = await mySession.run(feeds, fetches);
...
-If you don't want to use pre-allocated MLBuffer tensors for outputs, you can also specify the output data location in the session options:
+If you don't want to use pre-allocated MLTensor tensors for outputs, you can also specify the output data location in the session options:
 
 ```js
 const mySessionOptions1 = {
   ...,
-  // keep all output data on MLBuffer
-  preferredOutputLocation: 'ml-buffer'
+  // keep all output data on MLTensor
+  preferredOutputLocation: 'ml-tensor'
```
Comment (fdwr) [inline docs/tutorials/web/ep-webnn.md:153]: ~~Is there also a cpp/js file somewhere that needs to be updated to check for this string?~~ Never mind - this is just the doc.

**Review by fdwr — APPROVED** [review]
👍

File: docs/tutorials/web/ep-webnn.md (hunk 1 above)
Comment (Honry) [reply docs/tutorials/web/ep-webnn.md:102]: Oh, but the implementation hasn't changed, I will follow up in a new PR.

**Comment by fdwr** [comment]
Merging as it's pretty uncontentious.

The broken link failure is in another unrelated document, `stable-diffusion-csharp.html`, not `ep-webnn.md`:

> - ./_site/docs/tutorials/csharp/stable-diffusion-csharp.html
Ran on 289 files!
> *  External link https://huggingface.co/runwayml/stable-diffusion-v1-5/tree/onnx failed: 401 No error

**Pull request merged**